libraries/script-engine/src/ScriptEngine.h

```diff
@@ -200,6 +208,28 @@ public slots:
     void doneRunning();
 
 protected:
+    void init();
+
+    bool evaluatePending() const { return _evaluatesPending > 0; }
+    void timerFired();
+    void stopAllTimers();
+    void stopAllTimersForEntityScript(const EntityItemID& entityID);
+    void refreshFileScript(const EntityItemID& entityID);
+
+    void setParentURL(const QString& parentURL) { _parentURL = parentURL; }
+
+    QObject* setupTimerWithInterval(const QScriptValue& function, int intervalMS, bool isSingleShot);
+    void stopTimer(QTimer* timer);
+
+    QHash<EntityItemID, RegisteredEventHandlers> _registeredHandlers;
+    void forwardHandlerCall(const EntityItemID& entityID, const QString& eventName, QScriptValueList eventHanderArgs);
+    Q_INVOKABLE void entityScriptContentAvailable(const EntityItemID& entityID, const QString& scriptOrURL, const QString& contents, bool isURL, bool success);
+
+    EntityItemID currentEntityIdentifier {}; // Contains the defining entity script entity id during execution, if any. Empty for interface script execution.
+    QUrl currentSandboxURL {}; // The toplevel url string for the entity script that loaded the code being executed, else empty.
+    void doWithEnvironment(const EntityItemID& entityID, const QUrl& sandboxURL, std::function<void()> operation);
+    void callWithEnvironment(const EntityItemID& entityID, const QUrl& sandboxURL, QScriptValue function, QScriptValue thisObject, QScriptValueList args);
+
```

Comment:
I just shuffled the methods and properties around here so that the methods would be grouped together.